docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst

```diff
@@ -0,0 +1,616 @@
++---------------------------------+--------------+----------------+---------+
+| ``begin`` StmtSequence_ ``end`` | BeginExpr_   | StmtSequence_  |         |
++---------------------------------+--------------+----------------+---------+
+| ``break`` «Expr_»?              | BreakStmt_   | ReturningStmt_ |         |
++---------------------------------+--------------+----------------+---------+
+| ``;``                           | EmptyStmt_   | Stmt_          |         |
++---------------------------------+--------------+----------------+---------+
+| ``END {`` StmtSequence_ ``}``   | EndBlock_    | StmtSequence_  |         |
++---------------------------------+--------------+----------------+---------+
+| ``next`` «Expr_»?               | NextStmt_    | ReturningStmt_ |         |
++---------------------------------+--------------+----------------+---------+
+| ``redo``                        | RedoStmt_    | Stmt_          |         |
++---------------------------------+--------------+----------------+---------+
+| ``retry``                       | RetryStmt_   | Stmt_          |         |
++---------------------------------+--------------+----------------+---------+
+| ``return`` «Expr_»?             | ReturnStmt_  | ReturningStmt_ |         |
++---------------------------------+--------------+----------------+---------+
+| ``undef`` «FNAME ``,``»+        | UndefStmt_   | Stmt_          |         |
++---------------------------------+--------------+----------------+---------+
+
+Calls
+~~~~~
+
++----------------------------+---------------------+----------------+-------------------------------+
+| Expression syntax          | CodeQL class        | Superclasses   | Remarks                       |
++============================+=====================+================+===============================+
+| Expr_ ``[`` Expr_ ``]``    | ElementReference_   | MethodCall_    |                               |
++----------------------------+---------------------+----------------+-------------------------------+
+| MethodName_ «Expr_ ``,``»* | MethodCall_         | Call_          |                               |
++----------------------------+---------------------+----------------+-------------------------------+
+| LhsExpr_ ``=`` Expr_       | SetterMethodCall_   | MethodCall_    |                               |
++----------------------------+---------------------+----------------+-------------------------------+
+| ``super``                  | SuperCall_          | MethodCall_    |                               |
++----------------------------+---------------------+----------------+-------------------------------+
+| ``yield`` «Expr_ ``,``»*   | YieldCall_          | Call_          |                               |
++----------------------------+---------------------+----------------+-------------------------------+
+| ``&IDENTIFIER``            | BlockArgument_      | Expr_          | Used as an argument to a call |
```

Comment:
In the rest of this article, you've kept the placeholders that you defined at the start in normal text, so we should probably follow the same course here:

```suggestion
| ``&``IDENTIFIER            | BlockArgument_      | Expr_          | Used as an argument to a call |
```